src/RevCPU.cc

```diff
@@ -158,22 +158,19 @@ RevCPU::RevCPU( SST::ComponentId_t id, const SST::Params& params )
   Mem->SetMaxHeapSize( maxHeapSize );
 
   // Load the binary into memory
-  // TODO: Use std::nothrow to return null instead of throwing std::bad_alloc
-  Loader = new RevLoader( Exe, Args, Mem, &output );
+  Loader = new( std::nothrow ) RevLoader( Exe, Opts->GetArgv(), Mem, &output );
```

Comment:
Passing `std::nothrow` to `new` in placement-new syntax causes it to return `nullptr` if the allocation fails, similar to how `malloc()` behaves, instead of throwing a `std::bad_alloc`, which is how `new` usually behaves.

The original code tested the result of `new` against `nullptr` without specifying `std::nothrow`, so it was never going to catch a failed allocation --  `new` would have just raised a `std::bad_alloc` exception and unwound the stack until a `catch` caught it. If nothing caught it, the program would abort.